evap/staff/templates/staff_course_comments_quick.html

```diff
@@ -0,0 +1,403 @@
+            if(direction === "left" && index > 0) {
+                index--;
+            } else if(direction === "right" && index < items.length) {
+                index++;
+            }
+
+            if(index < items.length) {
+                active = items[index];
+                slideLayer(2, direction, active);
+            } else {
+                active = null;
+                slideLayer(2, direction, slider.find(".alert"))
+            }
+        }
+
+        function slideLayer(layer, direction, element) {
+            // to preserve the vertical positions and heights during transition,
+            // the elements will be inactivated from bottom to top and activated from top to bottom
+
+            if(element && element.is(".active")) {
+                return;
+            }
+
+            let last = findActive(layer);
+            if(last.length !== 0) {
+                let reversed = direction === "left"? "right" : "left";
+                last.css({top: last.position().top, height: last.outerHeight()});
+                last.removeClass("active to-left to-right");
+                last.addClass("to-" + reversed);
+            }
+
+            if(layer > 0) {
+                if(active) {
+                    let reference = element.prevAll("[data-layer=" + (layer - 1) + "]").first();
+                    slideLayer(layer - 1, direction, reference);
+                } else {
+                    slideLayer(layer - 1, direction, null);
+                }
+            }
+
+            if(element !== null) {
+                element.removeClass("to-left to-right");
+                element.addClass("to-" + direction);
+                element.position();
+                element.addClass("active");
+            }
+        }
+
+        function findActive(layer) {
+            if(layer === 2) {
+                return slider.find("[data-layer=2].active, .alert.active");
+            } else {
+                return slider.find("[data-layer=" + layer + "].active");
+            }
+        }
+
+        function repeat(action) {
+            let answer = slider.find("[data-layer=2]");
+
+            if(action === "start" && answer.length === 0) {
+                slideLayer(2, "right", slider.find(".alert"));
+                return;
+            }
+
+            let reviewed = [];
+            let unreviewed = [];
+
+            answer.each(function() {
+                let item = $(this);
+                if(item.is("[data-review]")) {
+                    reviewed.push(item);
+                } else {
+                    unreviewed.push(item);
+                }
+            });
+
+            let newIndex;
+
+            if(action !== "all" && unreviewed.length) {
+                newIndex = reviewed.length;
+            } else {
+                newIndex = 0;
+            }
+
+            items = reviewed.concat(unreviewed);
+            let direction = newIndex > index? "right" : "left";
```

Comment:
Official code style or begin of ideology wars? :smiling_imp:

Reply:
we don't have an official code style (yet), but if we had one it would surprise me if leaving out the space before the `?` were part of it. in fact, it would surprise me if that were part of any style guide anywhere. i have never seen that before, have you?

(my opinion: it's the operator with the lowest precedence right after assignment operators and the comma operator. we use spaces around all the other operators (except for the unary ones like ++ or !, which have very high precedence), and i don't see why of all operators the low-precedence ternary operator should not get spaces around it. in addition, in the case of `newIndex > index? foo : bar` leaving out the space is actually misleading, because it makes it look like `newIndex > (index? foo : bar)`. the only way i would find this remotely acceptable is by removing all the other spaces as well, or adding parenthesis, both of which seem unnecessary compared to adding the space.)

i would like to take the opportunity and thank my parents for raising me into such a privileged position that instead of worrying about housing and food, i have time to discuss this kind of stuff. thanks, mom and dad!

Reply:
Well, while I'm thinking about it... I think it is my personal style developed when I didn't understand the operator yet, so I tried to interpret the question mark as the end of a question and the colon as else. Like `(Est-ce que) [question]? [option] : [option]`, so I guess that your argumentation makes more sense.

Also thanks a lot for your stamina, while I'm still not convinced of all your suggestions I try them and look what will happen and I can really see the improvements.

Reply:
thanks for developing such a large feature with us and for persevering in the face of seemingly endless reviews ;)

Reply:
> Even infinities tend to age.